src/test/java/uk/gov/companieshouse/api/accounts/controller/LoansToDirectorsControllerTest.java

```diff
@@ -0,0 +1,194 @@
+package uk.gov.companieshouse.api.accounts.controller;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import uk.gov.companieshouse.api.accounts.AttributeName;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.model.rest.smallfull.notes.loanstodirectors.LoansToDirectors;
+import uk.gov.companieshouse.api.accounts.service.impl.LoansToDirectorsServiceImpl;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseStatus;
+import uk.gov.companieshouse.api.accounts.utility.ApiResponseMapper;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class LoansToDirectorsControllerTest {
+
+    private static final String COMPANY_ACCOUNT_ID = "companyAccountId";
+
+    @Mock
+    private HttpServletRequest request;
+
+    @Mock
+    private Transaction transaction;
+
+    @Mock
+    private LoansToDirectors loansToDirectors;
+
+    @Mock
+    private LoansToDirectorsServiceImpl loansToDirectorsService;
+
+    @Mock
+    private ApiResponseMapper apiResponseMapper;
+
+    @InjectMocks
+    private LoansToDirectorsController loansToDirectorsController;
+
+    @Test
+    @DisplayName("Loans to directors create - successful")
+    void createLoansToDirectorsSuccessful() throws DataException {
+
+        when(request.getAttribute(AttributeName.TRANSACTION.getValue())).thenReturn(transaction);
+
+        ResponseObject responseObject = new ResponseObject(ResponseStatus.CREATED,
```

Comment:
Want to type cast this?
```
ResponseObject<LoansToDirectors> responseObject = new ResponseObject<>(ResponseStatus.CREATED, loansToDirectors);
```

It's not needed in this case, as such, though it's good practise.

Reply:
sure thing!